addons/mail/static/tests/discuss/core/public_web/sub_channels.test.js

```diff
@@ -192,3 +193,75 @@ test("mention suggestions in thread match channel restrictions", async () => {
     await contains(".o-mail-Composer-suggestion", { text: "Mitchell Admin" });
     await contains(".o-mail-Composer-suggestion", { text: "p1" });
 });
+
+test("sub channel is muted (opacity-50) when mute is active", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "General" });
+    await start();
+    await openDiscuss(channelId);
+    await contains(".o-mail-Discuss-threadName", { value: "General" });
+    await click("button[title='Threads']");
+    await click("button[aria-label='Create Thread']");
+    await contains(".o-mail-Discuss-threadName", { value: "New Thread" });
+    await contains(".o-mail-DiscussSidebarChannel-subChannel.opacity-50", { count: 0 });
+    await click("button[title='Notification Settings']");
+    await click("button:contains('Mute Conversation')");
+    await click("button:contains('Until I turn it back on')");
+    await contains(".o-mail-DiscussSidebarChannel-subChannel.opacity-50", { count: 1 });
+});
+
+test("sub channel is not shown when parent channel is muted", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "General" });
+    await start();
+    await openDiscuss(channelId);
+    await contains(".o-mail-Discuss-threadName", { value: "General" });
+    await click("button[title='Threads']");
+    await click("button[aria-label='Create Thread']");
+    await contains(".o-mail-Discuss-threadName", { value: "New Thread" });
+    await contains(".o-mail-DiscussSidebarChannel-subChannel", { count: 1 });
+    await click(".o-mail-DiscussSidebarChannel");
+    await click("button[title='Notification Settings']");
```

Comment:
`.o-mail-DiscussSidebarChanenl-subChannel` and `.o-mail-DiscussSidebarChannel` are not great classnames to use in test: they do not map well with the way the end-user identify these items on UI.

Prefer `.o-mail-DiscussSidebar-item:contains('General')` and `.o-mail-DiscussSidebar-item:contains("New Thread")`: you click on the discuss item with these texts that are rendered on screen.